internal/resolver/safe_config_selector_test.go

```diff
@@ -0,0 +1,169 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package resolver
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/internal/grpctest"
+	"google.golang.org/grpc/internal/serviceconfig"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+type fakeConfigSelector struct {
+	selectConfig func(RPCInfo) *RPCConfig
+}
+
+func (f *fakeConfigSelector) SelectConfig(r RPCInfo) *RPCConfig {
+	return f.selectConfig(r)
+}
+
+func (s) TestSafeConfigSelector(t *testing.T) {
```

Comment:
This test is not very easy to read. Lots of goroutines, lots of different time values which are used for waiting/sleeping. Could this be simplified?

Reply:
It does look like a lot to cover in a test. Maybe its just me who finds it hard to read this test :)